src/types.rs

```diff
@@ -163,13 +169,17 @@ impl Point {
     ///
     /// assert!(dist < 1e-1)
     /// ```
-    pub fn distance_to(&self, point: &Point) -> f64 {
-        ((self.x() - point.x()).powi(2) + (self.y() - point.y()).powi(2)).sqrt()
+    pub fn distance_to(&self, point: &Point<T>) -> f64 {
```

Comment:
If instead of using [`num::Num`](http://rust-num.github.io/num/num/trait.Num.html) everywhere, we used [`num::Float`](http://rust-num.github.io/num/num/trait.Float.html), we could make `distance_to` generic by utilizing [`num::Float::sqrt`](http://rust-num.github.io/num/num/trait.Float.html#tymethod.sqrt).